File: 2849-determine-if-a-cell-is-reachable-at-a-given-time/2849-determine-if-a-cell-is-reachable-at-a-given-time.cpp
class Solution {
public:
    bool isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int diff_x = abs(sx - fx);
        int diff_y = abs(sy - fy);

        if(diff_x == 0 && diff_y == 0) {
            return t != 1;
        }

        return diff_x <= t && diff_y <= t;
    }
};

File: 2849-determine-if-a-cell-is-reachable-at-a-given-time/2849-determine-if-a-cell-is-reachable-at-a-given-time.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int diffX = Math.abs(sx - fx);
        int diffY = Math.abs(sy - fy);

        if(diffX == 0 && diffY == 0) {
            return t != 1;
        }

        return diffX <= t && diffY <= t;
    }
}